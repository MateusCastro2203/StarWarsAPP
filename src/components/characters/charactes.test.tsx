import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import CharactersTitle from './Characters'; // Ajuste o caminho de importação conforme necessário

describe('CharactersTitle', () => {
  test('snapshot', () => {
    const tree = render(
      <CharactersTitle text="Test Character" onPress={() => {}} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('deve renderizar o texto fornecido', () => {
    const {getByText} = render(
      <CharactersTitle text="Test Character" onPress={() => {}} />,
    );
    expect(getByText('Test Character')).toBeTruthy();
  });

  test('deve chamar onPress quando pressionado', () => {
    const onPressMock = jest.fn();
    const screen = render(
      <CharactersTitle text="Test Character" onPress={onPressMock} />,
    );
    fireEvent.press(screen.getByText('Test Character'));
    expect(onPressMock).toHaveBeenCalled();
  });
});
