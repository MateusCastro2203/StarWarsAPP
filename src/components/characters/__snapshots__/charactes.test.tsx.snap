// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharactersTitle snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "borderColor": "white",
      "borderRadius": 5,
      "borderWidth": 1,
      "margin": 20,
      "opacity": 1,
      "padding": 10,
    }
  }
>
  <Text
    style={
      {
        "color": "white",
        "fontSize": 20,
        "letterSpacing": 2,
      }
    }
  >
    Test Character
  </Text>
</View>
`;
